### PR TITLE
Updating index.md and other pages

### PR DESCRIPTION
**Pull request by azarboon**
I've updated a number of pages to include latest changes / to inform readers about their choices and associated complexities.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by azarboon** [comment]
@geoffcline as per your advice, I've combined all my PRs into one. Kindly please review this. Thanks a lot for your constructive feedbacks and your support. I really appreciate it.

**Comment by geoffcline** [comment]
Thank you for combining them @azarboon !

File: content/reliability/docs/application.md [hunk 1]
```diff
@@ -309,7 +309,7 @@ If you operate multiple clusters, you can use a service mesh to enable cross-clu
 
 ## Observability 
 
-Observability is an umbrella term that includes monitoring, logging, and tracing. Microservices based applications are distributed by nature. Unlike monolithic applications where monitoring a single system is sufficient, in a distributed application architecture, you need to monitor each component’s performance. You can use cluster-level monitoring, logging, and distributed tracing systems to identify issues in your cluster before they disrupt your customers. 
+Observability is an umbrella term that includes monitoring, logging, and tracing. One of the fallacies of distributed computing states that observability is optional. Microservices based applications are distributed by nature. Unlike monolithic applications where monitoring a single system is sufficient, in a distributed application architecture, you need to monitor each component’s performance. You can use cluster-level monitoring, logging, and distributed tracing systems to identify issues in your cluster before they disrupt your customers.
```
Comment (svennam92) [inline content/reliability/docs/application.md:312]: I prefer to leave this sentence out

File: content/performance/performance.md [hunk 2]
```diff
@@ -145,12 +145,17 @@ Generally, schedulers ensure that pods are placed only on nodes that have suffic
...
 
   + While HPA and VPA scale the deployments and pods, [Cluster Autoscaler](https://github.com/kubernetes/autoscaler) will scale-out  and scale-in the size of the pool of worker nodes. It adjusts the size of a Kubernetes cluster based on the current utilization. Cluster Autoscaler increases the size of the cluster when there are pods that failed to schedule on any of the current nodes due to insufficient resources or when adding a new node would increase the overall availability of cluster resources. Please follow this [step by step](https://eksworkshop.com/scaling/deploy_ca/) guide to setup Cluster Autoscaler.  If you are using Amazon EKS on AWS Fargate, AWS Manages the control plane for you. 
 
      Please have a look at the reliability pillar for detailed information.
      
 #### Monitoring 
+
+Please have a look at Monitoring and Observability sections in ["Running highly-available application"](https://aws.github.io/aws-eks-best-practices/reliability/docs/application) for detailed information.
```
Comment (svennam92) [inline content/performance/performance.md:156]: This page is not published, can we remove it from the changes?

File: content/index.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Introduction
 Welcome to the EKS Best Practices Guides.  The primary goal of this project is to offer a set of best practices for day 2 operations for Amazon EKS. We elected to publish this guidance to GitHub so we could iterate quickly, provide timely and effective recommendations for variety of concerns, and easily incorporate suggestions from the broader community.  
 
+Before taking any of our recommendations, remember that in software engineering everything is a trade-off. Also, you need to pay more attention to "why" you are making a decision than "how" to do it.
```
Comment (svennam92) [inline content/index.md:4]: Can we please remove this addition? I'm not certain this adds significant value for the reader.

**Comment by svennam92** [comment]
@CodiumAI-Agent /improve

**Comment by svennam92** [comment]
@azarboon Thank you for this PR! There's a lot of great additions here. I've left comments on a couple things I'd rather not include. Please also review the suggestions from the Codium AI Agent.

**Comment by azarboon** [comment]
@svennam92 thanks for the constructive feedback. I've committed all the suggestions by you and Codium. Kindly please merge the PR.

**Comment by azarboon** [comment]
@svennam92 and @geoffcline I've already incorporated your suggested changes since two weeks ago. Can you please merge this?

**Comment by svennam92** [comment]
Please re-open this PR against the asciidoc format we've migrated to (folder: latest/bpg)

**Pull request closed without merging**